lib/dao.js

```diff
@@ -757,12 +757,12 @@ DataAccessObject.findOrCreate = function findOrCreate(query, data, options, cb)
 
         // options has precedence on model-setting
         if (options.validate === false) {
-          _findOrCreate(query, data, obj);
+          return _findOrCreate(query, data, obj);
```

Comment:
It's a bad practice to mix unrelated changes (new features and bug fixes) in the same commit/pull request. Imagine yourself several months later, trying to figure out why exactly this line was changed, and `git blame` pointing you to a commit adding `replaceOrCreate` functionality. I would be confused.

So, could you please submit this fix as a new pull request and include a unit-test to reproduce the problem? We should land this fix ASAP and back-port it to 2.x too.